pkg/ccl/backupccl/backup_test.go

```diff
@@ -1645,6 +1646,75 @@ INSERT INTO d.t VALUES ('hello'), ('howdy');
 	})
 }
 
+func TestBackupRestoreDuringUserDefinedTypeChange(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Protects typeChangeStarted.
+	var mu syncutil.Mutex
+	typeChangeStarted := make(chan struct{})
```

Comment:
Super-minor optional nit: if you wanted to make it a bit more explicit that this `typeChangeStarted` channel is protected under a mutex, you could do something like:
```
mu := struct {
  syncutil.Mutex
  typeChangeStarted chan struct{}
}
```
so that usages to the channel would be through `mu.typeChangeStarted`.